In [3]:
import netCDF4
from netCDF4 import Dataset
import sys, os

import numpy as np
import pandas as pd 

my_path_gn = os.path.abspath("./output")
link_1 = my_path_gn + "/test_output_v3.nc"
link_2 = '/home/mikhail/ML/Diploma_ocean/Data/Data_1_09.txt'

In [7]:
def MAE(link_1, link_2, init_a, out_a, flg=0):
    rootgrp_1 = Dataset(link_1, "r", format="NETCDF4")

    coord_1_x = np.array(rootgrp_1.variables['longitude'][init_a:out_a].tolist())
    coord_1_y = np.array(rootgrp_1.variables['latitude'][init_a:out_a].tolist())
    if flg == 0:
        rootgrp_2 = Dataset(link_2, "r", format="NETCDF4")
        coord_2_x = np.array(rootgrp_2.variables['longitude'][init_a:out_a].tolist())
        coord_2_y = np.array(rootgrp_2.variables['latitude'][init_a:out_a].tolist())
    else:
        file = open(link_2)
        coord_2_x = []
        coord_2_y = []
        for line in file:
            # print(line)
            line_str = line.split()
            coord_2_x.append(float(line_str[0]))
            coord_2_y.append(float(line_str[1]))
    coord_2_x = np.resize(np.array(coord_2_x), 500)
    coord_2_y = np.resize(np.array(coord_2_y), 500)

            
             
    MAE = np.mean(np.sqrt(np.square(np.abs(coord_2_x - coord_1_x)) +\
                          np.square(np.abs(coord_2_y - coord_1_y))))
    print(MAE)
    

In [10]:
MAE(link_1, link_2, 12000, 12500, flg=1)

10.999126107855032


## Метрики ##

Real data
MAE 11.0227
point_line_spill
500 точек
Data:
    file_list = 'Data/Glorys/BS_hourly/20250828_h-CMCC--RFVL-BSeas8-BLK-b20250909_an-sv14.00.nc',
            'Data/Glorys/BS_hourly/20250829_h-CMCC--RFVL-BSeas8-BLK-b20250909_an-sv14.00.nc',
             'Data/Glorys/BS_hourly/20250830_h-CMCC--RFVL-BSeas8-BLK-b20250909_an-sv14.00.nc',
             'Data/Glorys/BS_hourly/20250831_h-CMCC--RFVL-BSeas8-BLK-b20250909_an-sv14.00.nc',
             'Data/Glorys/BS_hourly/20250901_h-CMCC--RFVL-BSeas8-BLK-b20250916_an-sv14.00.nc',
             'Data/Glorys/BS_hourly/20250902_h-CMCC--RFVL-BSeas8-BLK-b20250916_an-sv14.00.nc'




Nural WenHai data
MAE 10.9991
point_line_spill
500 точек
if True:
    file_list = '/home/mikhail/ML/Diploma_ocean/Models/Models_data/out_file.nc'


